File: validAnagram/valid_anagram.py
# Given two strings s and t, return true if t
# is an anagram of s, and false otherwise

def is_anagram(s, t):
    if (len(s) != len(t)):
        return False

    dict_s = {}
    dict_t = {}

    for i in s:
        if i not in dict_s:
            dict_s[i] = 0
        dict_s[i] += 1

    for j in t:
        if j not in dict_t:
            dict_t[j] = 0
        dict_t[j] += 1

    if dict_s != dict_t:
        return False
    return True

File: arraysAndHashing/validAnagram/validAnagram.js
// Given two strings s and t, return true if t
// is an anagram of s, and false otherwise

var isAnagram = function(s, t) {
    if (s.length !== t.length) return false;

    let objS = {};
    let objT = {};

    for (let i = 0; i < s.length; i++) {
        objS[s[i]] = (objS[s[i]] || 0) + 1;
    };

    for (let j = 0; j< t.length; j++) {
        objT[t[j]] = (objT[t[j]] || 0) + 1;
    };

    for (let key in objS) {
        if (objS[key] !== objT[key]) return false;
    }

    return true;
};
